### PR TITLE
webadmin: add tabs to Options dialog

### PR DESCRIPTION
**Pull request by rszwajko**
Changes:
1. move existing user options to dedicated tabs to allow extending
   functionality in the future
2. create a General section with username and email (read only)

### CODE REVIEW

**Comment by rszwajko** [comment]
### Before

image

### After

image


image


image


image

**Review by sgratch — CHANGES_REQUESTED** [review]
The code looks good and the dialog looks great as well.
Few general comments:

- Dialog title and menu entry label should be changed to something like 'User or Account settings' or just  'Settings', instead of 'Edit options' and 'Options'. IMHO, worth using the same terminology like on the VM portal.
- Consider approaching this same dialog from the vertical navigation menu as well by adding another menu option e.g. Administration -> Settings.
The same dialog will be available from both menus but on the other hand, the same goes for events as well (events dialog and events drawer). 
This is an important dialog so it seems to me that adding that to the vertical navigation menu is better than leaving that under the upper masthead only. WDYT?
- General tab: the user's 'last name' option is missing
- Consider moving the 'User's Public Key' option to the general tab since there aren't any other console options for now so not sure that opening a dedicated 'Console' tab with one option is better. Once there will be more console options then we can always add more tabs.
- The same goes for future added session settings like 'default initial tab' and 'session timeout'. All those should be part of general section tab in my opinion.
Not sure about the 'Tables' tab.

- The issue mentioned here  https://bugzilla.redhat.com/show_bug.cgi?id=1950321 is reproduced again now.

**Comment by rszwajko** [comment]
> * Dialog title and menu entry label should be changed to something like 'User or Account settings' or just  'Settings', instead of 'Edit options' and 'Options'. IMHO, worth using the same terminology like on the VM portal.
> 

+1

> * Consider approaching this same dialog from the vertical navigation menu as well by adding another menu option e.g. Administration -> Settings.
>   The same dialog will be available from both menus but on the other hand, the same goes for events as well (events dialog and events drawer).
>   This is an important dialog so it seems to me that adding that to the vertical navigation menu is better than leaving that under the upper masthead only. WDYT?
>

+1
 
> * General tab: the user's 'last name' option is missing
>

[The original design](https://marvelapp.com/prototype/6a95795/screen/62675607) included only User Name so Last Name was skipped.
We could add Last Name (and other fields) however IMHO this doesn't add much value and would duplicate functionality in Administration -> Users (actually even Administration -> Users -> [details] -> General is not displaying first/last name).

> * Consider moving the 'User's Public Key' option to the general tab since there aren't any other console options for now so not sure that opening a dedicated 'Console' tab with one option is better. Once there will be more console options then we can always add more tabs.
> 

Public key takes quite some place in the UI (although it's a single prop) - especially if we plan to expand the General section in the next PRs.

> * The same goes for future added session settings like 'default initial tab' and 'session timeout'. All those should be part of general section tab in my opinion.
>

+1

>   Not sure about the 'Tables' tab.
>

Depends on our plans (refresh timer? reset button?).
 
> * The issue mentioned here  https://bugzilla.redhat.com/show_bug.cgi?id=1950321 is reproduced again now.
>

Works for me. Please provide more information about scenario.

File: frontend/webadmin/modules/gwt-common/src/main/resources/org/ovirt/engine/ui/common/CommonApplicationConstants.properties [hunk 1]
```diff
@@ -286,6 +287,7 @@ editInstanceImages=Edit
 editInterface=Edit
 editProfile=Edit
 elementName=Name
+emailUser=E-mail
```
[the remark this answers is not in the document]
Comment (rszwajko) [reply frontend/webadmin/modules/gwt-common/src/main/resources/org/ovirt/engine/ui/common/CommonApplicationConstants.properties:290]: This should be possible in Zanata.

**Comment by rszwajko** [comment]
### Screenshots after renaming to Account Settings

image


image

image

**Comment by sgratch** [comment]
> > * General tab: the user's 'last name' option is missing
> 
> [The original design](https://marvelapp.com/prototype/6a95795/screen/62675607) included only User Name so Last Name was skipped. We could add Last Name (and other fields) however IMHO this doesn't add much value and would duplicate functionality in Administration -> Users (actually even Administration -> Users -> [details] -> General is not displaying first/last name).

Oops, what I meant here is that both first name + last name are missing (for users that have those set for him).

This was mentioned on our design (copied from the jira ticket: "Username (+ first name, last name if exists) - read only").
We are duplicating the email info as well, but we still display that here just to show the user his own general info. It is not displayed as part of the users-general tab, but it's displayed as part of the grid.  
I think this info is nice thing to display (if set), but if you think it's redundant then I'm totally ok with that. Just please update the ticket details so that QE won't search for that.

> 
> > * Consider moving the 'User's Public Key' option to the general tab since there aren't any other console options for now so not sure that opening a dedicated 'Console' tab with one option is better. Once there will be more console options then we can always add more tabs.
> 
> Public key takes quite some place in the UI (although it's a single prop) - especially if we plan to expand the General section in the next PRs.

Right, but on the other hand, having tabs with only one property each is also not that great design. And I anticipate users will ask for more console settings just because they see this empty console tab. If general tab includes only email and ssh key (in case first/last name is omitted) then won't there be a place for session timeout and initial tab?

> 
> > Not sure about the 'Tables' tab.
> 
> Depends on our plans (refresh timer? reset button?).

refresh times - not sure we'll have time for that
reset button - I'll check the other PR. I don't think reset should be at the same tab as other settings since this reset all tabs....

> 
> > * The issue mentioned here  https://bugzilla.redhat.com/show_bug.cgi?id=1950321 is reproduced again now.
> 
> Works for me. Please provide more information about scenario.

Check this screen recording. Only when clicking outside the text field, the ok button is enabled.

Peek 2022-02-16 21-05

**Review by sgratch — COMMENTED** [review]
Please check the latest comment.

**Comment by rszwajko** [comment]
@sgratch 
> please update the ticket details so that QE won't search for that.
>
Done

> If general tab includes only email and ssh key (in case first/last name is omitted) then won't there be a place for session timeout and initial tab?
>

Please check the new version with only 2 tabs. Note that since General tab has mixed content the header labels are needed. 

image

**Review by sjd78 — APPROVED** [review]
The code changes are straight-forward and LGTM

Design changes are reasonable:
 - Navigation points make sense to open the model
 - More tabs as the organization makes sense, but more so if there are say 3 or more items on each tab
 - The fewer tabs with sections on the tab page (as with the latest change) makes sense when there are not many items.

**Review by sgratch — APPROVED** [review]
LGTM

**Pull request merged**